From cae23c53be4888db8676cdd1a4632adac10f083f Mon Sep 17 00:00:00 2001
From: Josh Poimboeuf <jpoimboe@redhat.com>
Date: Thu, 4 Sep 2014 18:54:31 -0400
Subject: [kernel] trace: Remove ftrace_stop/start() from reading the trace file

Message-id: <9aea07b19ddef845cbdd3c7d1434a0a713da354a.1409856554.git.jpoimboe@redhat.com>
Patchwork-id: 91252
O-Subject: [RHEL7.1 PATCH 12/23] tracing: Remove ftrace_stop/start() from reading the trace file
Bugzilla: 1113829
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1113829

commit 099ed151675cd1d2dbeae1dac697975f6a68716d
Author: Steven Rostedt (Red Hat) <rostedt@goodmis.org>
Date:   Tue Jun 24 23:50:09 2014 -0400

    tracing: Remove ftrace_stop/start() from reading the trace file

    Disabling reading and writing to the trace file should not be able to
    disable all function tracing callbacks. There's other users today
    (like kprobes and perf). Reading a trace file should not stop those
    from happening.

    Cc: stable@vger.kernel.org # 3.0+
    Reviewed-by: Masami Hiramatsu <masami.hiramatsu.pt@hitachi.com>
    Signed-off-by: Steven Rostedt <rostedt@goodmis.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/kernel/trace/trace.c b/kernel/trace/trace.c
index cf12eaa..89ea344 100644
--- a/kernel/trace/trace.c
+++ b/kernel/trace/trace.c
@@ -1285,7 +1285,6 @@ void tracing_start(void)
 
  arch_spin_unlock(&ftrace_max_lock);
 
- ftrace_start();
  out:
  raw_spin_unlock_irqrestore(&global_trace.start_lock, flags);
 }
@@ -1332,7 +1331,6 @@ void tracing_stop(void)
  struct ring_buffer *buffer;
  unsigned long flags;
 
- ftrace_stop();
  raw_spin_lock_irqsave(&global_trace.start_lock, flags);
  if (global_trace.stop_count++)
   goto out;
-- 
1.7.1